lib/token.js

```diff
@@ -248,6 +249,129 @@ function addFragmentListener(sdk, windowEl, timeout) {
   return deferred.promise.timeout(timeout || 120000, new AuthSdkError('OAuth flow timed out'));
 }
 
+function handleOAuthResponse(sdk, oauthParams, res) {
+  if (res['error'] || res['error_description']) {
+    throw new OAuthError(res['error'], res['error_description']);
+  }
+
+  if (res.state !== oauthParams.state) {
+    throw new AuthSdkError('OAuth flow response state doesn\'t match request state');
+  }
+
+  var tokenTypes = oauthParams.responseType;
+  var scopes = util.clone(oauthParams.scope);
+  var tokenDict = {};
+
+  if (res['id_token']) {
+    var jwt = sdk.idToken.decode(res['id_token']);
+    if (jwt.payload.nonce !== oauthParams.nonce) {
+      throw new AuthSdkError('OAuth flow response nonce doesn\'t match request nonce');
+    }
+
+    var clientId = oauthParams.clientId || sdk.options.clientId;
+    validateClaims(jwt.payload, sdk.options.url, clientId);
+
+    var idToken = {
+      idToken: res['id_token'],
+      claims: jwt.payload,
+      expiresAt: jwt.payload.exp,
+      scopes: scopes
+    };
+
+    if (Array.isArray(tokenTypes)) {
+      tokenDict['id_token'] = idToken;
+    } else {
+      return idToken;
+    }
+  }
+  
+  if (res['access_token']) {
+    var accessToken = {
+      accessToken: res['access_token'],
+      expiresAt: Number(res['expires_in']) + Math.floor(Date.now()/1000),
+      tokenType: res['token_type'],
+      scopes: scopes
+    };
+
+    if (Array.isArray(tokenTypes)) {
+      tokenDict['token'] = accessToken;
+    } else {
+      return accessToken;
+    }
+  }
+
+  if (!tokenDict['token'] && !tokenDict['id_token']) {
+    throw new AuthSdkError('Unable to parse OAuth flow response');
+  }
+
+  var tokens = [];
+
+  // Create token array in the order of the responseType array
+  for (var t = 0, tl = tokenTypes.length; t < tl; t++) {
+    var tokenType = tokenTypes[t];
+    tokens.push(tokenDict[tokenType]);
```

Comment:
Good catch. It's technically possible if they pass a tokenType we don't support in addition to `token` or `id_token`. I'll add a check.